kedro_azureml/datasets/folder_dataset.py

```diff
@@ -43,3 +59,87 @@ def __init__(
                 f"underlying dataset. Please remove '{VERSIONED_FLAG_KEY}' flag from "
                 f"the dataset definition."
             )
+
+    @property
+    def path(self) -> str:
+        # For local runs we want to replicate the folder structure of the remote dataset.
+        # Otherwise kedros versioning would version at the file/folder level and not the
+        # AzureML dataset level
+        if self._local_run:
+            return (
+                Path(self.folder)
+                / self._azureml_dataset
+                / self.resolve_load_version()
+                / Path(self._dataset_config[self._filepath_arg])
+            )
+        else:
+            return Path(self.folder) / Path(self._dataset_config[self._filepath_arg])
+
+    @property
+    def download_path(self) -> str:
+        # Because `is_dir` and `is_file` don't work if the path does not
+        # exist, we use this heuristic to identify paths vs folders.
+        if self.path.suffix != "":
```

Comment:
This is maybe not super robust, since you could have a folder with a `.` in the name? Isn't the only case we want to treat specially where `filepath == "."`? Then we could just test for that.

Reply:
I think we also want to handle the case where the Parquet partitions are in a subfolder in a folder dataset. Therefore handling both `filepath == "."` and `filepath == "random/subfolder"` or not?

Reply:
The two options I could think of were testing for a "." in the filename or using the suffix. Not sure there is a more robust way ...

Reply:
Again, this was because of the confusing about how downloading works. I see, not ideal, but we can keep it for now.